apkid/rules/elf/obfuscators.yara

```diff
@@ -0,0 +1,137 @@
+ * Public License version 3.0 as published by the Free Software Foundation
+ * and appearing in the file LICENSE.GPL included in the packaging of this
+ * file. Please visit http://www.gnu.org/copyleft/gpl.html and review the
+ * information to ensure the GNU General Public License version 3.0
+ * requirements will be met.
+ *
+ **/
+
+
+
+rule ollvm_v3_4 : obfuscator
+{
+  meta:
+    description = "Obfuscator-LLVM version 3.4"
+    info        = "https://github.com/obfuscator-llvm/obfuscator/wiki"
+    example     = "cd16ad33bf203dbaa9add803a7a0740e3727e8e60c316d33206230ae5b985f25"
+    // "Obfuscator-clang version 3.4 (tags/RELEASE_34/final) (based on LLVM 3.4svn)"
+
+  strings:
+    $clang_version = "Obfuscator-clang version 3.4"
+    $based_on      = "(based on LLVM 3.4"
+
+  condition:
+    all of them
+}
+
+
+rule ollvm_v3_6_1 : obfuscator
+{
+  meta:
+    description = "Obfuscator-LLVM version 3.6.1"
+    info        = "https://github.com/obfuscator-llvm/obfuscator/wiki"
+    example     = "d84b45856b5c95f7a6e96ab0461648f22ad29d1c34a8e85588dad3d89f829208"
+    // "Obfuscator-LLVM clang version 3.6.1 (tags/RELEASE_361/final) (based on Obfuscator-LLVM 3.6.1)"
+
+  strings:
+    $clang_version = "Obfuscator-LLVM clang version 3.6.1"
+    $based_on      = "(based on Obfuscator-LLVM 3.6.1)"
+
+  condition:
+    all of them
+}
+
+
+rule ollvm_v4_0 : obfuscator
+{
+  meta:
+    description = "Obfuscator-LLVM version 4.x"
+    info        = "https://github.com/obfuscator-llvm/obfuscator/wiki"
+    example     = "aaba570388d0fe25df45480ecf894625be7affefaba24695d8c1528b974c00df"
+    // "Obfuscator-LLVM clang version 4.0.1  (based on Obfuscator-LLVM 4.0.1)"
+
+  strings:
+    $clang_version = "Obfuscator-LLVM clang version 4"
+    $based_on      = "(based on Obfuscator-LLVM"
+
+  condition:
+    all of them
+}
+
+
+rule ollvm_v6_0 : obfuscator
+{
+  meta:
+    description = "Obfuscator-LLVM version 6.0"
+    info        = "https://github.com/obfuscator-llvm/obfuscator/wiki"
+    example     = ""
+    // "Obfuscator-LLVM clang version 6.0.0 (trunk) (based on Obfuscator-LLVM 6.0.0git-b9ea5776)"
+    // "Obfuscator-LLVM clang version 6.0.0 (trunk) (based on Obfuscator-LLVM 6.0.0)"
+
+  strings:
+    $clang_version = "Obfuscator-LLVM clang version 6"
+    $based_on      = "(based on Obfuscator-LLVM 6"
+    $strenc        = /datadiv_decode[0-9]{18,20}/
+
+  condition:
+    ($clang_version and $based_on) and not $strenc
+}
+
+rule ollvm_v6_0_strenc : obfuscator
+{
+  meta:
+    description = "Obfuscator-LLVM version 6.0 (string encryption)"
+    info        = "https://github.com/obfuscator-llvm/obfuscator/wiki"
+    example     = "f3a2e6c57def9a8b4730965dd66ca0f243689153139758c44718b8c5ef9c1d17"
+    // "Obfuscator-LLVM clang version 6.0.0 (trunk) (based on Obfuscator-LLVM 6.0.0git-b9ea5776)"
+    // "Obfuscator-LLVM clang version 6.0.0 (trunk) (based on Obfuscator-LLVM 6.0.0)"
+
+  strings:
+    $clang_version = "Obfuscator-LLVM clang version 6"
```

Comment:
Could these rules have trailing spaces? Anything to make it a little better than just a plain string.

Reply:
There's no trailing space after this specific example. Got the idea but I do not know if I could get more extra trailing spaces on my strings. Please help me if my blindness does not allow me to see more today.